examples/unconditional_image_generation/train_unconditional.py

```diff
@@ -659,7 +665,22 @@ def transform_images(examples):
 
             if epoch % args.save_model_epochs == 0 or epoch == args.num_epochs - 1:
                 # save the model
+                unet = accelerator.unwrap_model(model)
+
+                if args.use_ema:
+                    ema_model.store(unet.parameters())
+                    ema_model.copy_to(unet.parameters())
+
+                pipeline = DDPMPipeline(
+                    unet=unet,
+                    scheduler=noise_scheduler,
+                )
+
                 pipeline.save_pretrained(args.output_dir)
+
+                if args.use_ema:
+                    ema_model.restore(unet.parameters())
+
```

Comment:
We have to make a separate pipeline here because it's not guaranteed that the pipeline set on line 633 is ever set. It's very likely that most people calling this script are never setting the save_model_epocs or save_images_epochs and they're always the same default values